File: src/main/java/itmo/is/project/repository/module/ModuleRepository.java
package itmo.is.project.repository.module;

import itmo.is.project.model.module.Module;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.repository.NoRepositoryBean;

@NoRepositoryBean
public interface ModuleRepository<T extends Module<?>> extends JpaRepository<T, Integer> {
}

File: src/main/kotlin/itmo/isproject/repository/module/ModuleRepository.kt
package itmo.isproject.repository.module

import itmo.isproject.model.module.Module
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.repository.NoRepositoryBean

@NoRepositoryBean
interface ModuleRepository<T : Module<*>> : JpaRepository<T, Int>
